daemon/integration/integration_test.go

```diff
@@ -55,9 +56,8 @@ func TestMain(m *testing.M) {
 		log.Fatal("could not make server", err)
 	}
 
-	l, err := server.Listen()
 	go func() {
-		s.Serve(l)
+		s.ListenAndServe()
```

Comment:
I found this could lead to a race condition, because the test could start (and try to connect to the port) before this goroutine is able to listen. WDYT?